model/speedscope_profile.go

```diff
@@ -38,41 +40,373 @@ type SpeedscopeProfileEvent struct {
 	File     *SpeedscopeFile
 }
 
-// XXX This parsing and raw speedscope struct could move to a separate internal package. That move would help with SpeedscopeProfile name collison.
-// https://www.speedscope.app/file-format-schema.json
-// https://github.com/jlfwong/speedscope/blob/main/src/lib/file-format-spec.ts
+type SpeedscopeFile struct {
```

Comment:
Separate from this change, but we should move all this Speedscope parsing to a separate "package". Currently every capitalized identifier here is visible in the whole "mode" package.